localstack/services/cloudformation/models/opensearch.py

```diff
@@ -0,0 +1,82 @@
+from localstack.aws.api.opensearch import (
+    OpenSearchPartitionInstanceType,
+    OpenSearchWarmPartitionInstanceType,
+)
+from localstack.services.cloudformation.deployment_utils import remove_none_values
+from localstack.services.cloudformation.service_models import GenericBaseModel
+from localstack.utils.aws import aws_stack
+from localstack.utils.collections import select_attributes
+
+
+# OpenSearch still uses "es" ARNs
+# See examples in:
+# https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-opensearchservice-domain.html
+def opensearch_add_tags_params(params, **kwargs):
+    es_arn = aws_stack.es_domain_arn(params.get("DomainName"))
+    tags = params.get("Tags", [])
+    return {"ARN": es_arn, "TagList": tags}
+
+
+class OpenSearchDomain(GenericBaseModel):
+    @staticmethod
+    def cloudformation_type():
+        return "AWS::OpenSearchService::Domain"
+
+    def get_physical_resource_id(self, attribute=None, **kwargs):
+        domain_name = self._domain_name()
+        if attribute == "Arn":
+            # As mentioned above, OpenSearch still uses "es" ARNs
+            return aws_stack.elasticsearch_domain_arn(domain_name)
+        return domain_name
+
+    def fetch_state(self, stack_name, resources):
+        domain_name = self._domain_name()
+        domain_name = self.resolve_refs_recursively(stack_name, domain_name, resources)
+        return aws_stack.connect_to_service("opensearch").describe_domain(DomainName=domain_name)
+
+    def _domain_name(self):
+        return self.props.get("DomainName") or self.resource_id
+
+    @staticmethod
+    def get_deploy_templates():
+        def _create_params(params, **kwargs):
+            attributes = [
+                "AccessPolicies",
+                "AdvancedOptions",
+                "CognitoOptions",
+                "DomainName",
+                "EBSOptions",
+                "ClusterConfig",
+                "EngineVersion",
+                "EncryptionAtRestOptions",
+                "LogPublishingOptions",
+                "NodeToNodeEncryptionOptions",
+                "SnapshotOptions",
+                "VPCOptions",
+            ]
```

Comment:
This list of attributes is missing some of the possible attributes: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-opensearchservice-domain.html

Maybe you could directly use the generated type from our ASF API: https://github.com/localstack/localstack/blob/7474d65cbb680289a65a21f6ee83232cf9be52f2/localstack/aws/api/opensearch/__init__.py#L750-L766
You could access the members with `CreateDomainRequest.__annotations__`, but it seems that the request slightly differs from the CF resource (f.e. the `CreateDomainRequest` contains `TagList` while the CF template has a `Tags` member).

Reply:
As far as I've understood it, the tags get handled via the `add_tags` operation in the `create` list of the return value of `get_deploy_templates`. So, omitting that value from the ASF-based list would suffice.